File: backend.Domain/Entities/Ayah.cs
namespace backend.Domain.Entities;

public class Ayah
{
    public int Id { get; set; }
    public int SurahId { get; set; }
    public Surah Surah { get; set; } = null!;
    public int AyahNumber { get; set; }
    public string ArabicText { get; set; } = string.Empty;
    public string NormalizedArabicText { get; set; } = string.Empty;
    public string EnglishTranslation { get; set; } = string.Empty;
    public string BanglaTranslation { get; set; } = string.Empty;
    public string Transliteration { get; set; } = string.Empty;

    public int? Page { get; set; }
    public int? Juz { get; set; }
    public int? Manzil { get; set; }
    public int? Ruku { get; set; }
    public int? HizbQuarter { get; set; }
    public bool? Sajda { get; set; }
}

File: android/app/src/main/java/com/example/quran_app/domain/model/Ayah.kt
package com.example.quran_app.domain.model

import com.google.gson.annotations.SerializedName

data class Ayah(
    @SerializedName("id") val id: Int,
    @SerializedName("surahId") val surahId: Int,
    @SerializedName("ayahNumber") val ayahNumber: Int,
    @SerializedName("arabicText") val arabicText: String,
    @SerializedName("englishTranslation") val englishTranslation: String,
    @SerializedName("banglaTranslation") val banglaTranslation: String,
    @SerializedName("transliteration") val transliteration: String,
    @SerializedName("page") val page: Int?,
    @SerializedName("juz") val juz: Int?,
    @SerializedName("manzil") val manzil: Int?,
    @SerializedName("ruku") val ruku: Int?,
    @SerializedName("hizbQuarter") val hizbQuarter: Int?,
    @SerializedName("sajda") val sajda: Boolean?,
    @SerializedName("audioUrl") val audioUrl: String = "",
)
